File: app/src/main/java/com/justcode/hxl/viewutil/自定义View/绘制1_1/饼图/interfaces/SectorFactory.java
package com.justcode.hxl.viewutil.自定义View.绘制1_1.饼图.interfaces;

import android.support.annotation.NonNull;

import com.justcode.hxl.viewutil.自定义View.绘制1_1.饼图.drawable.BaseSectorDrawable;
import com.justcode.hxl.viewutil.自定义View.绘制1_1.饼图.entry.PieEntry;


public interface SectorFactory <T extends PieEntry,K extends BaseSectorDrawable> {
    K createSector(@NonNull T pieEntry, int position);
}

File: app/src/main/java/com/justcode/hxl/viewutil/自定义View/绘制1_1/饼图/interface/SectorFactory.kt
package com.justcode.hxl.viewutil.自定义View.绘制1_1.饼图.`interface`

import com.justcode.hxl.viewutil.自定义View.绘制1_1.饼图.drawable.BaseSectorDrawable
import com.justcode.hxl.viewutil.自定义View.绘制1_1.饼图.entry.PieEntry

interface SectorFactory<T : PieEntry, K : BaseSectorDrawable> {
    fun createSector(pieEntry: T, position: Int): K
}